lib/axiom/fs/stream/readable_stream_forwarder.js

```diff
@@ -24,14 +25,23 @@ var ReadableStream;
  */
 export var ReadableStreamForwarder = function(stream) {
   AxiomStream.call(this);
-  /** type {!AxiomEvent} */
+ 
+  /** @const @type {!ReadableStream} */
+  this.stream = stream;
+
+  /** @const @type {!AxiomEvent} */
   this.onData = new AxiomEvent();
-  /** type {!AxiomEvent} */
+  /** @const @type {!AxiomEvent} */
   this.onReadable = new AxiomEvent();
-  /** @type {!ReadableStream} */
-  this.stream = stream;
+  /** @const @type {!AxiomEvent} */
+  this.onEnd = new AxiomEvent();
+  /** @const @type {!AxiomEvent} */
+  this.onClose = new AxiomEvent();
+  
   this.stream.onData.addListener(this.onDataCallback_, this);
   this.stream.onReadable.addListener(this.onReadableCallback_, this);
+  this.stream.onEnd.listenOnce(this.end, this);
```

Comment:
Oh I see,nevermind then, I had mis-read the code. I thought this was pointing to a custom event handler function, not an already existing one